main.go

```diff
@@ -12,429 +13,264 @@ import (
-	// Get flags from the cli
 	hidecpu := pflag.Bool("nocpu", false, "Hide the CPU info")
 	hidemem := pflag.Bool("nomem", false, "Hide the RAM info")
 	hidetemp := pflag.Bool("notemp", false, "Hide the Temperature info")
 	hidefans := pflag.Bool("nofans", false, "Hide the Fans info")
 
-	// Get arguments from the cli
-	mode := pflag.StringSlice("mode", []string{"basic"}, "Output modes to enable {basic|ports|poe}")
-	hostName = pflag.StringP("hostname", "H", "http://192.168.112.7", "Hostname to use")
-	username := pflag.StringP("username", "u", "", "Username to use for authentication")
-	password := pflag.StringP("password", "p", "", "Password to use for authentication")
-
-	// Warning & Critical values for every metric
-	CPU_WARN := pflag.Float64("cpu-warning", 50, "Provide the minimum for CPU usage warning")
-	CPU_CRIT := pflag.Float64("cpu-critical", 90, "Provide the minimum for CPU usage critical")
-	MEM_WARN := pflag.Float64("mem-warning", 50, "Provide the minimum for RAM usage warning")
-	MEM_CRIT := pflag.Float64("mem-critical", 90, "Provide the minimum for RAM usage critical")
-	FAN_WARN := pflag.Float64("fan-warning", 3000, "Provide the minimum for Fan speeds warning")
-
-	TEMP_WARN := pflag.Float64("temp-warning", 50, "Provide the minimum for Temperature warning")
-	TEMP_CRIT := pflag.Float64("temp-critical", 70, "Provide the minimum for Temperature critical")
-	STATS_WARN := pflag.Float64("stats-warning", 5, "Provide the minimum for Port statistics warning")
-	STATS_CRIT := pflag.Float64("stats-critical", 20, "Provide the minimum for Port statistics critical")
-
-	// Ports to check if the mode 'ports' is present
-	portsToCheck := pflag.IntSlice("port", []int{1, 2, 3, 4, 5, 6, 7, 8}, "Ports to check")
+	mode := pflag.StringSlice("mode", []string{"basic"}, "Output modes to enable {basic|ports|poe|all}")
+	hostName := pflag.StringP("hostname", "H", "http://192.168.112.7", "Hostname to use")
+	username := pflag.StringP("username", "u", "", "Username for authentication")
+	password := pflag.StringP("password", "p", "", "Password for authentication")
+
+	// Thresholds
+	CPU_WARN := pflag.Float64("cpu-warning", 50, "CPU usage warning threshold")
+	CPU_CRIT := pflag.Float64("cpu-critical", 90, "CPU usage critical threshold")
+	MEM_WARN := pflag.Float64("mem-warning", 50, "RAM usage warning threshold")
+	MEM_CRIT := pflag.Float64("mem-critical", 90, "RAM usage critical threshold")
+	FAN_WARN := pflag.Float64("fan-warning", 3000, "Fan speed warning threshold")
+	TEMP_WARN := pflag.Float64("temp-warning", 50, "Temperature warning threshold")
+	TEMP_CRIT := pflag.Float64("temp-critical", 70, "Temperature critical threshold")
+	STATS_WARN := pflag.Float64("stats-warning", 5, "Port stats warning threshold")
+	STATS_CRIT := pflag.Float64("stats-critical", 20, "Port stats critical threshold")
 
+	portsToCheck := pflag.IntSlice("port", []int{1, 2, 3, 4, 5, 6, 7, 8}, "Ports to check")
 	help := pflag.BoolP("help", "h", false, "Show this help")
```

Comment:
This might be a bit far fetched, but why are you using `github.com/spf13/pflag` and not [Go's `flag`](https://pkg.go.dev/flag)? So far, I have not seen anything fancy requiring such a fancy library.

Reply:
The fancy thing that this library supported was that it maps the long version `--help` and shorthand `-h` to the same argument while the standard library doesn't support such. But I think, for now the shorthand form can be dropped entirely as I commented here https://github.com/Icinga/check-netgear/pull/6#discussion_r2503796860